Sources/Leaf/LeafEncoder.swift

```diff
@@ -131,11 +123,8 @@ private final class KeyedContainer<Key>: KeyedEncodingContainerProtocol, _Contai
     func encode<T>(_ value: T, forKey key: Key) throws
         where T : Encodable
     {
-        if let convertible = value as? LeafDataRepresentable {
-            guard let converted = convertible.leafData else {
-                throw EncodingError.invalidValue(value, at: self.codingPath + [key])
-            }
-            self.dictionary[key.stringValue] = .data(converted)
+        if let leafRepresentable = value as? LeafDataRepresentable {
+            self.dictionary[key.stringValue] = .data(leafRepresentable.leafData)
```

Comment:
Remove conditional assignment from change to `LeafDataRepresentable.leafData: LeafData`